soh/src/overlays/actors/ovl_En_Hy/z_en_hy.c

```diff
@@ -659,7 +659,17 @@ s16 func_80A70058(GlobalContext* globalCtx, Actor* thisx) {
                     gSaveContext.dogParams = 0;
                     break;
                 case 0x709F:
-                    func_80A6F7CC(this, globalCtx, (gSaveContext.infTable[25] & 2) ? GI_RUPEE_BLUE : gSaveContext.n64ddFlag ? Randomizer_GetItemIdFromKnownCheck(RC_MARKET_LOST_DOG, GI_HEART_PIECE) : GI_HEART_PIECE);
+                    if (gSaveContext.infTable[25] & 2) { // Already brought the lost dog back
+                        func_80A6F7CC(this, globalCtx, GI_RUPEE_BLUE);
+                    } else {
+                        if (!gSaveContext.n64ddFlag) {
+                            func_80A6F7CC(this, globalCtx, GI_HEART_PIECE);
+                        } else {
+                            GetItemEntry getItemEntry = Randomizer_GetItemFromKnownCheck(RC_MARKET_LOST_DOG, GI_HEART_PIECE);
+                            this->unkGetItemId = getItemEntry.getItemId;
+                            GiveItemEntryFromActor(&this->actor, globalCtx, getItemEntry, this->actor.xzDistToPlayer + 1.0f, fabsf(this->actor.yDistToPlayer) + 1.0f);
```

Comment:
This call looks different to the non-rando calls above, what is `func_80A6F7CC` doing differently? Just setting those distance values?

Reply:
Yeah, `func_80A6F7CC` just does
```c
this->unkGetItemId = getItemId;
func_8002F434(&this->actor, globalCtx, getItemId, this->actor.xzDistToPlayer + 1.0f,
                  fabsf(this->actor.yDistToPlayer) + 1.0f);
```

So I copied them out here, I just added a comment explaining they were copied